pkg/component/autoscaling/gardenercustommetrics/deployment.go

```diff
@@ -0,0 +1,137 @@
+	return map[string]string{
+		v1beta1constants.LabelApp:   "gardener-custom-metrics",
+		v1beta1constants.GardenRole: "gardener-custom-metrics",
+	}
+}
+
+func (gcmx *gardenerCustomMetrics) deployment(serverSecretName string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: gcmx.namespace,
+			Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+				resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeServer,
+			}),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             ptr.To[int32](1),
+			RevisionHistoryLimit: ptr.To[int32](2),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: getLabels(),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+						v1beta1constants.LabelNetworkPolicyToDNS:                                   v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer:                      v1beta1constants.LabelNetworkPolicyAllowed,
+						"networking.resources.gardener.cloud/to-all-shoots-kube-apiserver-tcp-443": v1beta1constants.LabelNetworkPolicyAllowed,
+					}),
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Args: []string{
+								"--secure-port=6443",
+								"--tls-cert-file=/var/run/secrets/gardener.cloud/tls/tls.crt",
+								"--tls-private-key-file=/var/run/secrets/gardener.cloud/tls/tls.key",
+								"--leader-election=true",
+								"--namespace=" + gcmx.namespace,
+								"--access-ip=$(POD_IP)",
+								"--access-port=6443",
+								"--log-level=74",
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "POD_IP",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "status.podIP",
+										},
+									},
+								},
+								{
+									Name: "LEADER_ELECTION_NAMESPACE",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.namespace",
+										},
+									},
+								},
+							},
+							Image:           gcmx.values.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Name:            gcmxContainerName,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 6443,
+									Name:          "metrics-server",
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("80m"),
+									corev1.ResourceMemory: resource.MustParse("200Mi"),
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									MountPath: "/var/run/secrets/gardener.cloud/tls",
+									Name:      "gardener-custom-metrics-tls",
```

Comment:
Extract this as a constant and use in the volume definition

Reply:
Ah, I see... You must be referring to the mount name. What I extracted is the mount path, which also had potential to deduplicate.

Reply:
Volume name is also deduplicated now.